src/OmnichannelChatSDK.ts

```diff
@@ -2042,30 +2043,47 @@ class OmnichannelChatSDK {
                     this.scenarioMarker.completeScenario(TelemetryEvent.GetAuthToken);
                 } else {
                     const exceptionDetails = {
-                        response: "UndefinedAuthToken"
+                        response: ChatSDKErrorName.UndefinedAuthToken
                     };
 
+                    if (optionalParams?.throwError) {
+                        throw Error(exceptionDetails.response);
+                    }
+
+                    // Fail scenario only if error is not thrown then bubbled up
                     this.scenarioMarker.failScenario(TelemetryEvent.GetAuthToken, {
                         ExceptionDetails: JSON.stringify(exceptionDetails)
                     });
                 }
-            } catch {
+            } catch (error) {
                 const exceptionDetails = {
-                    response: "GetAuthTokenFailed"
+                    response: ChatSDKErrorName.GetAuthTokenFailed as string
                 };
 
+                if ((error as Error).message == ChatSDKErrorName.UndefinedAuthToken) {
+                    exceptionDetails.response = (error as Error).message;
+                }
+
                 this.scenarioMarker.failScenario(TelemetryEvent.GetAuthToken, {
                     ExceptionDetails: JSON.stringify(exceptionDetails)
                 });
+
+                if (optionalParams?.throwError) {
+                    throw Error(exceptionDetails.response);
+                }
             }
         } else {
             const exceptionDetails = {
-                response: "GetAuthTokenNotFound"
+                response: ChatSDKErrorName.GetAuthTokenNotFound
             };
 
             this.scenarioMarker.failScenario(TelemetryEvent.GetAuthToken, {
                 ExceptionDetails: JSON.stringify(exceptionDetails)
             });
+
+            if (optionalParams?.throwError) {
```

Comment:
why we want to silent exceptions?,

Reply:
It's for backward compatibility. We have two options to set the `authToken`

1. During ChatSDK initialization
```js
const chatSDKConfig = {
  getAuthToken: async () => {return "token"}
}

const chatSDK = new OmnichannelChatSDK(omnichannelConfig, chatSDKConfig); // `getAuthToken()` is passed then executed there
```
2. After ChatSDK initialization, but before chat start
```js
const chatSDK = new OmnichannelChatSDK(omnichannelConfig); // `getAuthToken()` is not passed but would not thrown any exception 
await chatSDK.initialize();
await chatSDK.setAuthTokenProvider(async () => "token");
await chatSDK.startChat(); // exception would be thrown there if we're not able to retrieve `authToken`
```

Default users usually go for option **1** . Option **2** is used in `lcw` only due to design restrictions.

`getAuthToken()` could fail due to the auth token provider service not available intermittently so we'd have retry in start chat in case this happens.